### PR TITLE
initial changes to reduce number of byte allocations

### PR DESCRIPTION
**Pull request by Icehunter**

### CODE REVIEW

**Comment by Icehunter** [comment]
@ravahn how does this look? :P

File: Sharlayan/Models/Signature.cs [hunk 1]
```diff
@@ -87,9 +91,15 @@ public IntPtr GetAddress()
             {
                 return baseAddress;
             }
-            return MemoryHandler.Instance.ResolvePointerPath(PointerPath, baseAddress, IsASMSignature);
+            if (_resolvedPointerPath == IntPtr.Zero)
+            {
+                _resolvedPointerPath = MemoryHandler.Instance.ResolvePointerPath(PointerPath, baseAddress, IsASMSignature);
```
Comment (ravahn) [inline Sharlayan/Models/Signature.cs:96]: Is it safe to cache the resolved pointer path? What if the memory address is reallocated? for example, when zoning, addresses?  Not sure how you handle Chatlog pointers but they do change when the buffer becomes full.

Comment (Icehunter) [reply Sharlayan/Models/Signature.cs:96]: Um... good question. For chatlog pointers (at least offsets) I ensure indexes on each scan of the chatlog array. But the initial offset I deal with shouldn't change on zoning though??? At least it never has for me.

Did you find it changes?

**Review by ravahn — CHANGES_REQUESTED** [review]
All looks good except possibly the caching of resolved pointers.  just review that to make sure it is resilient to changes in game memory (i.e. doesn't assume intermediate pointers / final pointers are static for the lifetime of the FFXIV process).  You should also test in-game by zoning/etc.

File: Sharlayan/Models/Signature.cs (hunk 1 above)
Comment (ravahn) [reply Sharlayan/Models/Signature.cs:96]: I didn't test.  probably just roll this one change back since the rest is good?

**Comment by Icehunter** [comment]
@FFXIVAPP/maintainers this is already taken care of for the most part in other PR's that have been merged. Cached pointers is not used.

**Pull request closed without merging**